core/services/pipeline/task_params.go

```diff
@@ -233,7 +233,7 @@ func (b *BytesParam) UnmarshalPipelineParam(val interface{}) error {
 	switch v := val.(type) {
 	case string:
 		// try hex first
-		if len(v) >= 2 && v[:2] == "0x" {
```

Comment:
This did not take into account capital case (`0X`) variant.

Reply:
We probably want to avoid this unless necessary.